app/controllers/profiles_controller.rb

```diff
@@ -0,0 +1,25 @@
+class ProfilesController < ApplicationController
+  before_action :require_authentication
+
+  def show; end
+
+  def update
+    return unless current_user?
```

Comment:
Don't need this now you have the `require_authentication` check above.

Reply:
I added that in originally as an extra layer, but have removed it in the latest commit